### PR TITLE
Use GLib for parsing .desktop files

### PR DESCRIPTION
**Pull request by thp**
This aligns better with other Desktop platforms and removes
all the custom code for parsing locales and .desktop files.

Also remove Spectacle cruft, as our build-dependencies changed.

### CODE REVIEW

**Comment by pvuorela** [comment]
.yaml file removal in glib commit, should be in the cleanups. Having both things changing quite a few lines, could suggest merging separately. LGTM already at this point for the cleanup parts.

**Comment by thp** [comment]
Split cleanups into separate #23 and removed the YAML changes from this commit.

File: src/mdesktopentry.cpp [hunk 1]
```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
...
+        return d_ptr->desktopEntriesMap.value(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.value(key);
 }
 
 QString MDesktopEntry::value(const QString &group, const QString &key) const
 {
-    return d_ptr->desktopEntriesMap.value(group + '/' + key);
+    if (group != DesktopEntrySection) {
+        return QString();
```
Comment (pvuorela) [inline src/mdesktopentry.cpp]: Appears this doesn't anymore support other groups than the required Desktop Entry. Current version, as far as I see, does read all the groups.

File: src/mdesktopentry.cpp [hunk 2]
```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
 
 bool MDesktopEntry::contains(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.contains(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.contains(key);
```
Comment (pvuorela) [inline src/mdesktopentry.cpp]: If only Desktop Entry content in read into the map, this and similar should fail?

File: tests/ut_mdesktopentry.cpp [hunk 3]
```diff
@@ -396,6 +395,7 @@ void UtMDesktopEntry::localization()
 
     if (!lang.isEmpty()) {
         qputenv("LANG", lang.toLocal8Bit());
+        qputenv("LC_ALL", lang.toLocal8Bit());
```
Comment (pvuorela) [inline tests/ut_mdesktopentry.cpp]: Problems somewhere with only LANG? Could expect that to be enough.

Comment (thp) [inline tests/ut_mdesktopentry.cpp]: It was enough for the old code (that did the `$LANG` parsing manually), but with glib, I had failing tests with `LC_ALL=de_AT.UTF-8` from the client environment.

Comment (thp) [inline tests/ut_mdesktopentry.cpp]: And to quote SUSv2:

<q>     This variable determines the values for all locale categories. The value of the LC_ALL environment variable has precedence over any of the other environment variables starting with LC_ (LC_COLLATE, LC_CTYPE, LC_MESSAGES, LC_MONETARY, LC_NUMERIC, LC_TIME) and the LANG environment variable. </q>

(http://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html)

File: src/mdesktopentry.cpp (hunk 1 above)
Comment (thp) [inline src/mdesktopentry.cpp]: Yes, that's true. If we decide to support all groups, I'd suggest keeping the `GKeyFile` around and not filling `desktopEntriesMap` with all entries, of which for the usual case only a few of the Desktop entry are kept.

Do we know of any code that actually uses different sections/keys from a .desktop file via `MDesktopEntry`?

File: src/mdesktopentry.cpp (hunk 2 above)
Comment (thp) [inline src/mdesktopentry.cpp]: True. Let's see below if we want to have other content or not, I'm open for both options.

File: src/mdesktopentry.cpp (hunk 1 above)
Comment (pvuorela) [inline src/mdesktopentry.cpp]: I'm not aware of uses, but if aim is to enhance standard compliance on few details, would consider it good not to regress on others :) Also these group based query methods are nonsensical with support for only one group. 
Ditching desktopEntriesMap sounds good.

Comment (rburchell) [inline src/mdesktopentry.cpp]: Keep in mind that it is also public API, so you should be quite careful with regards to behaviour changes, as we can't fix everything using it

File: tests/ut_mdesktopentry.cpp (hunk 3 above)
Comment (pvuorela) [inline tests/ut_mdesktopentry.cpp]: Yea, LC_ALL should have precedence over any other (which should make LANG above redundant too), but code should still work with LANG only, being used if no proper LC_* is found. Jolla uses only LANG for setting device locale.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**